src/lsm/segmented_array.zig

```diff
@@ -139,11 +198,15 @@ fn SegmentedArrayType(
                 absolute_index: u32,
                 elements: []const T,
             ) void {
-                return array.insert_elements_at_absolute_index(
+                if (options.verify) array.verify();
+
+                array.insert_elements_at_absolute_index(
                     node_pool,
                     absolute_index,
                     elements,
                 );
+
+                if (options.verify) array.verify();
```

Comment:
What do you think of this instead? (I'm not sure which is nicer):
```suggestion
                if (options.verify) array.verify();
                defer if (options.verify) array.verify();

                return array.insert_elements_at_absolute_index(
                    node_pool,
                    absolute_index,
                    elements,
                );
```

Reply:
I've been burned by using defer in this way before, because it will still trigger if you throw an error. 

(There is an issue for `noerrdefer` that was closed as resolved with no comment indicating the resolution...)

Reply:
Ooh, good call. Yes, let's leave it as-is.